database/002_initial_schema.sql

```diff
@@ -1,5 +1,470 @@
+
+
+-- Extension
+
+CREATE EXTENSION citext;
+
+-- Domain 
+CREATE DOMAIN email AS citext
+  CHECK (value ~ '^[a-zA-Z0-9.!#$%&''*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$' );
+
+-- ENUMS
+
+CREATE TYPE statues AS ENUM ('0', '1', '2');
+CREATE TYPE Etypes AS ENUM('itex','booking','ad');
+-- CONFIG
+REVOKE CONNECT ON DATABASE booking_dev FROM PUBLIC;
+REVOKE ALL ON ALL TABLES IN SCHEMA public FROM PUBLIC;
+
+-- Tables
+
+-- Company
+
+CREATE TABLE
+IF NOT EXISTS public.company(
+    CID SERIAL PRIMARY KEY,
+    email citext UNIQUE NOT NULL,
+    orgNum bigint UNIQUE NOT NULL,
+    username text NOT NULL ,
+    salt text NOT NULL ,
+    passhash varchar NOT NULL ,
+    fullName text UNIQUE NOT NULL
+);
+
+-- Company User
+CREATE TABLE
+IF NOT EXISTS public.companyuser(
+    CUID SERIAL PRIMARY KEY,
+    email citext UNIQUE,
+    telephone text,
+    username text NOT NULL,
+    salt text NOT NULL,
+    passhash varchar NOT NULL ,
+    fullName text UNIQUE NOT NULL,
```

Comment:
The reason it's unique is more an implementation detail rather that something that should not be changed. When i made the system i assumed we would set the use the same username and password for the dbuser as for the user on the server. But we could set it so that instead of the fullname we use the username instead, then i could remove the unique constraint on fullname and split it into given and surname.